packages/context/src/binding.ts

```diff
@@ -95,6 +96,24 @@ export class Binding {
   }
 
   /**
+   * Bind the key to a MetaValueProvider
+   * @param providerClass {Function} The MetaValueProvider class (constructor function)
+   * @example
+   * ```ts
+   * ctx.bind('now').toProvider(MyProvider);
+   *
+   * ```
+   */
+  public toProvider<T>(providerClass: Constructor<MetaValueProvider>): this {
+    this.valueConstructor = providerClass;
+    this.getValue = (ctx): Object | Function => {
+      const providerInstance: MetaValueProvider = instantiateClass<MetaValueProvider>(providerClass, ctx) as MetaValueProvider;
```

Comment:
`instantiateClass<T>` returns `T | Promise<T>`, you need to handle both cases here. 

Something along the following lines:

```ts
const providerOrPromise = instantiateClass<MetaValueProvider>(providerClass, ctx);
if (isPromise(providerOrPromise)) {
  return providerOrPromise.then(p => p.value());
} else {
  return p.value();
}
```

Please add a test for the promise case.